tokio/src/signal/mod.rs

```diff
@@ -58,3 +60,45 @@ mod os {
 
 pub mod unix;
 pub mod windows;
+
+// Privately "vendor" the reusable_future mod without copying the code
+// FIXME: is this a good idea or should we copy the module?
+#[allow(dead_code, unreachable_pub)]
+#[path = "../../../tokio-util/src/sync/reusable_box.rs"]
+mod reusable_box;
```

Comment:
The source of `tokio-util` will not be available when publishing to [crates.io][1]. You could consider just using an ordinary pinned box instead.

[1]: https://crates.io/

Reply:
Or "just" copy the code across.

Reply:
Honestly, for signals, a plain old pinned box is probably fine.

Reply:
Oh yeah good point, will fix